libsql-server/src/connection/connection_manager.rs

```diff
@@ -302,24 +313,12 @@ impl WrapWal<Sqlite3Wal> for ManagedConnectionWalWrapper {
     fn end_read_txn(&mut self, wrapped: &mut Sqlite3Wal) {
         wrapped.end_read_txn();
         {
-            let mut current = self.manager.current.lock();
+            let current = self.manager.current.lock();
             if let Some((id, _, true)) = *current {
+                // releasing read transaction releases the write lock (see wal.c)
                 if id == self.id {
-                    let next = loop {
-                        match self.manager.write_queue.steal() {
-                            Steal::Empty => break None,
-                            Steal::Success(item) => break Some(item),
-                            Steal::Retry => (),
-                        }
-                    };
-
-                    if let Some((id, unparker)) = next {
-                        tracing::debug!(line = line!(), "unparking id={id}");
-                        *current = Some((id, Instant::now(), false));
-                        unparker.unpark()
-                    } else {
-                        *current = None;
-                    }
+                    drop(current);
+                    self.release();
```

Comment:
Is it a problem if a race condition happens between releasing and getting the current lock?

Reply:
unless the slot is empty, it's only replaced on release, and only a connection is allowed to release itself, so it's safe :)